sql/src/main/scala/org/apache/spark/sql/sedona_sql/io/geoparquet/ParquetFileFormat.scala

```diff
@@ -0,0 +1,545 @@
+/*
```

Comment:
Looks like all these are duplicated with Spark internal parquet reader implementation. From my understanding the geoparquet is just one specific type of parquet with its only schema. 
Generally, what differs here? how about implementing a very lean layer on top of spark native parquet file reader using DataSourceV2 interface with specific schema?

Reply:
@jiayuasu if I start working on geoparquet predicate pushdown, will it be invalidated if you are moving to the datasource V2 api soon? Is there a branch with the v2 api ("[We have a neat version that nicely supports Spark 3.3 only](https://github.com/apache/incubator-sedona/pull/652)")?

Reply:
@neontty My understanding is that the merged PR in Sedona already uses Datasource v2 as Spark Parquet reader is using it. But Spark parquet reader is designed in a way that most critical functions are private, so we have to copy lots of their code for adding a few changes on our own.

The version that supports Spark 3.3 only is here: https://github.com/ashar236/incubator-sedona/tree/geoparquet-spark3.3/sql/src/main/scala/org/apache/spark/sql/execution/datasources/parquet

In short, I believe your geoparquet predicate pushdown function will Not be invalidated. But I would suggest that please minimize your change to the codebase.